lib/pbench/agent/tool_data_sink.py

```diff
@@ -124,25 +124,78 @@ def log_request(self, code="-", size="-"):
 
         self.options["handler_class"] = DataSinkWsgiRequestHandler
         self._server = None
+        self._err_code = None
+        self._err_text = None
         self._lock = Lock()
         self._cv = Condition(lock=self._lock)
         self._logger = logger
 
     def run(self, app):
+        """run - Start the WSGI server, called by the Bottle framework.
+
+        Intended to be run as a separate thread.
+
+        We record the outcome of the `make_server` call for success or failure
+        and notify anybody waiting for this thread to succeed.
+        """
         assert self._server is None, "'run' method called twice"
         self._logger.debug("Making tool data sink WSGI server ...")
-        server = make_server(self.host, self.port, app, **self.options)
-        with self._lock:
-            self._server = server
-            self._cv.notify()
-        self._logger.debug("Running tool data sink WSGI server ...")
-        self._server.serve_forever()
+        try:
+            server = make_server(self.host, self.port, app, **self.options)
+        except OSError as exc:
+            assert exc.errno != 0, "Logic bomb!  OSError exception with no errno value"
+            with self._lock:
```

Comment:
We don't bother logging the exception because we are passing the exception string and the `errno` value to the waiting thread, which will handle the behavior from there.  We only log the general exception because we don't know what kind of state the caller will be in.